internal/elasticsearch/transform/transform.go

```diff
@@ -0,0 +1,548 @@
+			Optional:     true,
+			Default:      "30s",
+			ValidateFunc: utils.StringIsDuration,
+		},
+		"enabled": {
+			Type:        schema.TypeBool,
+			Description: "Controls wether the transform is started or stopped. Default is `false` (stopped).",
+			Optional:    true,
+			Default:     false,
+		},
+	}
+
+	utils.AddConnectionSchema(transformSchema)
+
+	return &schema.Resource{
+		Schema:      transformSchema,
+		Description: "Manages Elasticsearch transforms. See: https://www.elastic.co/guide/en/elasticsearch/reference/current/transforms.html",
+
+		CreateContext: resourceTransformCreate,
+		ReadContext:   resourceTransformRead,
+		UpdateContext: resourceTransformUpdate,
+		DeleteContext: resourceTransformDelete,
+	}
+}
+
+func resourceTransformCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	client, diags := clients.NewApiClient(d, meta)
+	if diags.HasError() {
+		return diags
+	}
+
+	transformName := d.Get("name").(string)
+	id, diags := client.ID(ctx, transformName)
+	if diags.HasError() {
+		return diags
+	}
+
+	transform, err := getTransformFromResourceData(ctx, d, transformName)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	params := models.PutTransformParams{
+		DeferValidation: d.Get("defer_validation").(bool),
+	}
+
+	timeout, err := time.ParseDuration(d.Get("timeout").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	params.Timeout = timeout
+
+	params.Enabled = d.Get("enabled").(bool)
+
+	if diags := elasticsearch.PutTransform(ctx, client, transform, &params); diags.HasError() {
+		return diags
+	}
+
+	d.SetId(id.String())
+	return resourceTransformRead(ctx, d, meta)
+}
+
+func resourceTransformRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	client, diags := clients.NewApiClient(d, meta)
+	if diags.HasError() {
+		return diags
+	}
+	compId, diags := clients.CompositeIdFromStr(d.Id())
+	if diags.HasError() {
+		return diags
+	}
+
+	transformName := compId.ResourceId
+	if err := d.Set("name", transformName); err != nil {
+		return diag.FromErr(err)
+	}
+
+	transform, diags := elasticsearch.GetTransform(ctx, client, &transformName)
+	if transform == nil && diags == nil {
+		tflog.Warn(ctx, fmt.Sprintf(`Transform "%s" not found, removing from state`, compId.ResourceId))
+		d.SetId("")
+		return diags
+	}
+	if diags.HasError() {
+		return diags
+	}
+
+	return diags
+}
+
+func resourceTransformUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	client, diags := clients.NewApiClient(d, meta)
+	if diags.HasError() {
+		return diags
+	}
+
+	transformName := d.Get("name").(string)
+	_, diags = client.ID(ctx, transformName)
+	if diags.HasError() {
+		return diags
+	}
+
+	updatedTransform, err := getTransformFromResourceData(ctx, d, transformName)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	updatedTransform.Pivot = nil
+	updatedTransform.Latest = nil
```

Comment:
Is this required by the API or is this just added safety? It looks like these are marked as `ForceNew` so Terraform would handle the delete/create if there were changes.

I'm not suggesting a change here, just curious :)

Reply:
It's API-related. I'm using the same struct (model), marshaled into a json payload, for both Create and Update API calls. However, as per Elastic API specs, there is a small difference: properties `pivot` and `latest` are not expected on Update operations, even if I were to send the exact same `pivot` or `latest` as on creation.
The alternative would have been to use 2 models, one for Create, one for Update, but that felt like overkill, given how similar they are.